### PR TITLE
add a batch client

### PR DESCRIPTION
**Pull request by shenghaozou**
Some option I provide:
--ip: set ip
--port: set port
--project: set project which is used for batch grading
--num_containers: considering that running docker may consume a lot of resources, you can set how many requests I will send at the same time. By default it is 2 but I've tried 8 in local machine and it works well. I will wait for 4 seconds after I send num_containers of request.
--num_tasks: only for debugging. set how many requests in total I will send. e.g. only 10 so I can easily debug for the code.

### CODE REVIEW

**Comment by shenghaozou** [comment]
Hi Tyler, I have added late day fetching logic.
The last column is the late day. I use -1 to represent NULL. (It happens when students don't submit anything.)

File: dockerGrader/client/dockerGraderClient.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
+import argparse, boto3, botocore, csv, math, logging, requests, threading, time, json
+
+REQUEST_URL="http://{ip}:{port}/json/{project}/{netId}"
+RESULT_PATH="ta/grading/{project}/{netId}.json"
+METADATA_PATH="projects/{project}/users/{googleId}/curr.json"
```
Comment (tylerharter) [inline dockerGrader/client/dockerGraderClient.py]: This is their full submission, not just metadata.  When I was generating meta.json from here before, I was downloading from this location, but then I was popping the payload (which contains b64 student code) to leave only metadata.

File: dockerGrader/client/dockerGraderClient.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            "Error: {}".format(errorReason) if errorReason else "",
+            lateDays))
+        gradeInfo.append((netId, grade, errorReason, intLateDays))
+    return gradeInfo
+
+def generateCsv(gradeInfo, saveFileName):
+    with open(saveFileName, "w") as csvfw:
+        csvWriter = csv.writer(csvfw)
+        for netId, grade, errorReason, lateDays in gradeInfo:
+            csvWriter.writerow([netId, grade, errorReason, lateDays])
```
Comment (tylerharter) [inline dockerGrader/client/dockerGraderClient.py]: can we generate a row for their partner too?

**Comment by tylerharter** [comment]
Sounds good, thanks!  Just sent you some minor comments, but feel free to
merge in when you're ready.

**Comment by tylerharter** [comment]
Is this one ready to merge?  I thought you fixed it so there are different directories under tmp (so maybe it's ok that it's always result.json...

**Comment by shenghaozou** [comment]
Yes.

**Pull request merged**